lib/url-pattern/url-match-pattern.js

```diff
@@ -0,0 +1,279 @@
+var _ = require('../util').lodash,
+    Property = require('../collection/property').Property,
+    Url = require('../collection/url').Url,
+
+    MATCH_ALL = '*',
+    PREFIX_DELIMITER = '^',
+    POSTFIX_DELIMITER = '$',
+    MATCH_ALL_URLS = '<all_urls>',
+    ALLOWED_PROTOCOLS = ['http', 'https', 'file', 'ftp'],
+    ALLOWED_PROTOCOLS_REGEX = ALLOWED_PROTOCOLS.join('|'),
+
+    regexes = {
+        escapeMatcher: /[.+^${}()|[\]\\]/g,
+        escapeMatchReplacement: '\\$&',
+        questionmarkMatcher: /\?/g,
+        questionmarkReplacment: '.',
+        starMatcher: '*',
+        starReplacement: '.*',
+        patternSplit: '^(' + ALLOWED_PROTOCOLS_REGEX + '|\\*)*://(\\*|\\*\\.[^*/]+|[^*/]+|)(/.*)$'
+    },
+
+    UrlMatchPattern;
+
+/**
+ * UrlMatchPattern allows to create rules to define Urls to match for.
+ * It is based on Google's Match Pattern - https://developer.chrome.com/extensions/match_patterns
+ *
+ * @example <caption>An example UrlMatchPattern</caption>
+ * var matchPattern = new UrlMatchPattern('https://*.google.com/*');
+*/
+_.inherit((
+  UrlMatchPattern = function UrlMatchPattern (pattern) {
+      UrlMatchPattern.super_.apply(this, { pattern: pattern });
```

Comment:
Doing .apply with an object?? Also, why do we need to change the arguments while calling super class. Not a good idea.

The way to handle when you want to allow "string" type arguments is to add `parse` and `unparse` to the instance. See how this is handled in headers - https://github.com/postmanlabs/postman-collection/blob/develop/lib/collection/header.js